src/ape/managers/networks.py

```diff
@@ -31,27 +30,46 @@ class NetworkManager:
     plugin_manager: PluginManager
     active_provider: Optional[ProviderAPI] = None
     _default: Optional[str] = None
+    _project_ecosystems: Dict[str, Dict[str, EcosystemAPI]] = {}
 
     def __repr__(self):
         return f"<NetworkManager, active_provider={self.active_provider}>"
 
-    @cached_property
+    @property
```

Comment:
Don't be fooled! The laziness goes on here still. Only now, we have to re-initialize the Ecosystem classes when we change project directories (as we do in the tests). This way, the new config file changes can get applied to the providers (such as the default provider!)